ext/symengine/symengine_utils.c

```diff
@@ -28,6 +30,31 @@ void sympify(VALUE operand2, basic_struct *cbasic_operand2) {
             basic_free_stack(den_basic);
             break;
 
+        case T_COMPLEX:
+            real = rb_funcall(operand2, rb_intern("real"), 0, NULL);
+            imag = rb_funcall(operand2, rb_intern("imaginary"), 0, NULL);
+
+            basic_struct *real_basic = basic_new_heap();
+            basic_struct *imag_basic = basic_new_heap();
+
+            sympify(real, real_basic);
+            sympify(imag, imag_basic);
+
+            
+            basic_struct *const_I = basic_new_heap();
+            basic_struct *imag_part = basic_new_heap();
+
+            basic_const_I(const_I);
+            basic_mul(imag_part, const_I, imag_basic);
+            basic_add(cbasic_operand2, imag_part, real_basic);
+
+            basic_free_heap(real_basic);
+            basic_free_heap(imag_basic);
+            basic_free_heap(const_I);
+            basic_free_heap(imag_part);
```

Comment:
Since you don't need to these `basic_struct`s to exist even when the function completes, you should use basic_new_stack. They will be faster too.